Cargo.toml

```diff
@@ -36,3 +40,80 @@ gtk_3_16 = ["gtk_3_10", "gtk/v3_16"]
 gtk_3_18 = ["gtk_3_16", "gtk/v3_18"] #for CI tools
 gtk_3_20 = ["gtk_3_18", "gtk/v3_20"] #for CI tools
 gtk_3_22 = ["gtk_3_20", "gtk/v3_22"] #for CI tools
+futures = ["futures-core", "futures-executor", "futures-channel", "futures-util", "glib/futures", "gio/futures"]
+
+[patch.'https://github.com/gtk-rs/gio']
+gio = { git = 'https://github.com/sdroege/gio-rs', branch = 'futures' }
```

Comment:
This currently patches in my GIO branch from https://github.com/gtk-rs/gio/pull/108. Will remove once that is merged